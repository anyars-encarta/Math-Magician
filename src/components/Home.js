import React from 'react';
import styles from '../modules/Home.module.css';
import GitHub from '../assets/GitHub.png';
import Linkedin from '../assets/Linkedin.png';
import Twitter from '../assets/Twitter.png';
import Facebook from '../assets/Facebook.png';
import YouTube from '../assets/YouTube.png';

function Home() {
  return (
    <div className={styles.home}>
      <h2>Welcome to our page!</h2>
      <p className={styles.p}>
        Encarta Math Magicians is an app developed by Anyars Yussif, a Full
        Stack Software Development student at Microverse. Microverse is an
        online school that uses peer-programming and project building to teach.
        <br />
        <br />
        Encarta Maths Magicians is a Calculator app that performs arithmetic calculations.
        The page has been devided into three sections:&nbsp;
        <strong>Home</strong>
        ,&nbsp;
        <strong>Calculator</strong>
        , and&nbsp;
        <strong>Quote</strong>
        .
        <br />
        <br />
        The Home section displays basic information about the author as well as what
        the app does. The Calculator section displays the calculator which you can
        use to perform arithmetic calculations. The Quote sections displays quotes
        fetched from API Ninjas. Every displayed quote refreshes after every 15 seconds
        to display a new quote.
        <br />
        <br />
        Have fun using&nbsp;
        <strong>Encarta Math Magicians!</strong>
        <br />
        <br />
        Follow the Author on:
      </p>
      <br />

      <div className={styles.followContainer}>
        <a className={styles.follow} href="https://github.com/anyars-encarta" target="_blank" rel="noreferrer">
          <img src={GitHub} alt="GitHub logo" />
        </a>
        <br />
        <a className={styles.follow} href="https://www.linkedin.com/in/anyars-yussif-1a179769/" target="_blank" rel="noreferrer">
          <img src={Linkedin} alt="Linkedin logo" />
        </a>
        <br />
        <a className={styles.follow} href="https://twitter.com/anyarsencarta" target="_blank" rel="noreferrer">
          <img src={Twitter} alt="Twitter logo" />
        </a>
        <br />
        <a className={styles.follow} href="https://web.facebook.com/encartadegreat?_rdc=1&_rdr" target="_blank" rel="noreferrer">
          <img src={Facebook} alt="Facebook logo" />
        </a>
        <br />
        <a className={styles.follow} href="https://www.youtube.com/channel/UC8ROvrdEomu6xYI0HrzT61A" target="_blank" rel="noreferrer">
          <img src={YouTube} alt="Youtube logo" />
        </a>
      </div>
    </div>
  );
}

export default Home;
