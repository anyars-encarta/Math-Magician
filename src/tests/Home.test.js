// Home.test.js
import React from 'react';
import { render, screen } from '@testing-library/react';
import Home from '../components/Home';

test('renders Home component correctly', () => {
  render(<Home />);
  expect(screen.getByText('Welcome to our page!')).toBeInTheDocument();
  expect(screen.getByText(/Encarta Math Magicians is an app developed by Anyars Yussif, a Full Stack Software Development student at Microverse. Microverse is an online school that uses peer-programming and project building to teach./)).toBeInTheDocument();
});

test('matches Home component snapshot', () => {
  const { asFragment } = render(<Home />);
  expect(asFragment()).toMatchSnapshot();
});
