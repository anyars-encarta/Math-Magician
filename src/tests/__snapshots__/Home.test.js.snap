// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Home component snapshot 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <h2>
      Welcome to our page!
    </h2>
    <p
      class="p"
    >
      Encarta Math Magicians is an app developed by Anyars Yussif, a Full Stack Software Development student at Microverse. Microverse is an online school that uses peer-programming and project building to teach.
      <br />
      <br />
      Encarta Maths Magicians is a Calculator app that performs arithmetic calculations. The page has been devided into three sections: 
      <strong>
        Home
      </strong>
      , 
      <strong>
        Calculator
      </strong>
      , and 
      <strong>
        Quote
      </strong>
      .
      <br />
      <br />
      The Home section displays basic information about the author as well as what the app does. The Calculator section displays the calculator which you can use to perform arithmetic calculations. The Quote sections displays quotes fetched from API Ninjas. Every displayed quote refreshes after every 15 seconds to display a new quote.
      <br />
      <br />
      Have fun using 
      <strong>
        Encarta Math Magicians!
      </strong>
      <br />
      <br />
      Follow the Author on:
    </p>
    <br />
    <div
      class="followContainer"
    >
      <a
        class="follow"
        href="https://github.com/anyars-encarta"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="GitHub logo"
          src="GitHub.png"
        />
      </a>
      <br />
      <a
        class="follow"
        href="https://www.linkedin.com/in/anyars-yussif-1a179769/"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Linkedin logo"
          src="Linkedin.png"
        />
      </a>
      <br />
      <a
        class="follow"
        href="https://twitter.com/anyarsencarta"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Twitter logo"
          src="Twitter.png"
        />
      </a>
      <br />
      <a
        class="follow"
        href="https://web.facebook.com/encartadegreat?_rdc=1&_rdr"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Facebook logo"
          src="Facebook.png"
        />
      </a>
      <br />
      <a
        class="follow"
        href="https://www.youtube.com/channel/UC8ROvrdEomu6xYI0HrzT61A"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Youtube logo"
          src="YouTube.png"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
